crates/uv-client/src/middleware.rs

```diff
@@ -45,3 +47,73 @@ impl Middleware for OfflineMiddleware {
         ))
     }
 }
+
+/// A middleware with support for netrc files.
+///
+/// Based on: <https://github.com/gribouille/netrc>.
+pub(crate) struct NetrcMiddleware {
+    nrc: Netrc,
+}
+
+impl NetrcMiddleware {
+    pub(crate) fn new() -> Result<Self> {
+        Netrc::new().map(|nrc| NetrcMiddleware { nrc })
+    }
+}
+
+#[async_trait::async_trait]
+impl Middleware for NetrcMiddleware {
+    async fn handle(
+        &self,
+        mut req: Request,
+        _extensions: &mut Extensions,
+        next: Next<'_>,
+    ) -> reqwest_middleware::Result<Response> {
+        if let Some(auth) = req.url().host_str().and_then(|host| {
+            self.nrc
+                .hosts
+                .get(host)
+                .or_else(|| self.nrc.hosts.get("default"))
+        }) {
+            // Remove the existing `Authorization` header, if any.
+            req.headers_mut().remove(reqwest::header::AUTHORIZATION);
```

Comment:
Does this mean for a URL with explicit auth attached we will replace it with what's in the netrc file? That seems wrong.

Reply:
Yes. I believe this is correct actually.

Reply:
`pip` changed this in https://github.com/pypa/pip/pull/10998. It was then reverted in https://github.com/pypa/pip/pull/11134.

Reply:
Why? Wouldn't it be very confusing if you passed authentication via the CLI and we overwrote it with something from a configuration file?

Reply:
Or maybe not? I honestly can't tell: https://github.com/pypa/pip/blob/a33caa26f2bf525eafb4ec004f9c1bd18d238a31/src/pip/_internal/network/auth.py#L347. Anyway, I can switch the precedence if you prefer. Doesn't matter to me.

Reply:
pip's behavior is more complicated because they only attach auth after seeing a 401 — it looks like they changed that for netrc? I'm not sure I didn't give it a close read yet.

I think it's important to respect _directly_ passed credentials over the persistent configuration.

Reply:
I changed it.

Reply:
I wish there was more context on that revert pull request...

Reply:
but the issue it closes is https://github.com/pypa/pip/issues/11113

Reply:
Yeah it seems like pip's _current_ intent is to prioritize the URL over the netrc. This does make sense to me, I was just trying to follow pip and misread the chain of reverts. Thanks!